airflow-core/src/airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -342,9 +342,9 @@ def get_dag_runs(
     query = select(DagRun)
 
     if dag_id != "~":
-        dag: DAG = dag_bag.get_dag(dag_id)
+        dag: DAG | None = get_dag_from_dag_bag(dag_bag, dag_id)
         if not dag:
-            raise HTTPException(status.HTTP_404_NOT_FOUND, f"The DAG with dag_id: `{dag_id}` was not found")
+            raise HTTPException(status.HTTP_404_NOT_FOUND, f"Dag with id {dag_id} was not found")
```

Comment:
Please leave the original message, this is more consistent with how other 404 messages are constructed.